core/store/models/log_events.go

```diff
@@ -423,10 +426,26 @@ func (parseRunLog20190207withoutIndexes) parseJSON(log Log) (JSON, error) {
 		return JSON{}, err
 	}
 
+	// 8 bytes worth of data versions is plenty.
+	b := bytes.NewBuffer(log.Data[expirationEnd+versionSize-8 : expirationEnd+versionSize])
```

Comment:
Can this ever panic with out of bounds if Data is of an unexpected size?

Reply:
There is a `SafeByteSlice` data type to protect against this sort of thing, I might suggest taking a look at that

Reply:
no - there's a length check up front https://github.com/smartcontractkit/chainlink/blob/08caa9f7de4ce0cd251d6e2c2793ec56fed452d3/core/store/models/log_events.go#L400

Reply:
Can we use SafeByteSlice anyway? Belts and braces.

Reply:
done